pkg/validatorfactory/factory.go

```diff
@@ -369,7 +369,14 @@ func (s *ValidatorFactory) ValidatorsForGVK(gvk schema.GroupVersionKind) (*Valid
 			continue
 		}
 
-		val := newValidatorEntry(nam, namespaced.Has(gvk), def, ssf)
+		// Try to infer the scope from paths
+		nsScoped := namespaced.Has(gvk)
+		// Check schema extensions to see if the scope was manually added
+		if scope, ok := def.Extensions.GetString("x-kubectl-validate-scope"); ok {
+			nsScoped = scope == strings.ToLower(string(apiextensions.NamespaceScoped))
```

Comment:
use lowercase version of both strings for comparison (considering case where users custom schemas use this to communicate the scope, not just internal code)

Reply:
> considering case where users custom schemas use this to communicate the scope, not just internal code

what does that mean ?

Reply:
> considering case where users custom schemas use this to communicate the scope, not just internal code

i switched to lowercase version of both strings for comparison but not sure to understand the sentence above